srai/loaders/osm_loaders/osm_tile_data_collector.py

```diff
@@ -0,0 +1,100 @@
+"""This module contains classes of strategy for handling downloaded tiles."""
+from abc import ABC, abstractmethod
+from enum import Enum
+from pathlib import Path
+from typing import Any, Union
+
+from PIL import Image
+
+
+class DataCollector(ABC):
+    """Stores collected images."""
+
+    @abstractmethod
+    def store(self, idx: str, data: Image.Image) -> Any:
+        """
+        Apply action for object storage and returns data of it.
+
+        Args:
+            idx (str): id of tile
+            data (Image.Image): tile
+        """
+
+
+class SavingDataCollector(DataCollector):
+    """
+    Saves in disk collected images.
+
+    Stores paths.
+    """
+
+    def __init__(self, save_path: Union[str, Path], file_extension: str) -> None:
+        """
+        Initialize SavingDataCollector.
+
+        Args:
+            save_path (Union[str, Path]): root path for data
+            file_extension (str): file name extension
+        """
+        super().__init__()
+        if save_path is None or file_extension is None:
+            raise ValueError
+        self.save_path = Path(save_path)
+        self.format = file_extension
+
+    def store(self, idx: str, data: Image.Image) -> Path:
+        """
+        Saves image on disk. Returns path.
```

Comment:
We try to stick with the imperative form.
```suggestion
        Save image on disk. Returns path.
```